include/mega/filter.h

```diff
@@ -0,0 +1,190 @@
+/**
+ * @file mega/filter.h
+ * @brief Classes representing file filters.
+ *
+ * (c) 2013-2014 by Mega Limited, Auckland, New Zealand
+ *
+ * This file is part of the MEGA SDK - Client Access Engine.
+ *
+ * Applications using the MEGA API must present a valid application key
+ * and comply with the the rules set forth in the Terms of Service.
+ *
+ * The MEGA SDK is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * @copyright Simplified (2-clause) BSD License.
+ *
+ * You should have received a copy of the license along with this
+ * program.
+ */
+#ifndef MEGA_FILTER_H
+#define MEGA_FILTER_H 1
+
+#include "types.h"
+
+namespace mega
+{
+
+// Forward Declarations (from Filesystem)
+struct MEGA_API FileAccess;
+struct MEGA_API InputStreamAccess;
+
+enum FilterStrategy
+{
+    FS_GLOB,
+    FS_REGEX,
+    NUM_FILTER_STRATEGIES
+}; /* FilterStrategy */
+
+enum FilterTarget
+{
+    FT_ALL,
+    FT_DIRECTORIES,
+    FT_FILES,
+    NUM_FILTER_TARGETS
+}; /* FilterTarget */
+
+enum FilterType
+{
+    FT_NAME,
```

Comment:
Should we find a different prefix, so it doesn't clash with the prefix for values in `FilterTarget`?

Reply:
This branch / PR can be removed.

The changes that could be integrated, have been.

feature/ignorefile has changed a fair bit since this PR was created.
These enumerations no longer exist, for example.